prog/postgres/postgres_server_nexus.rb

```diff
@@ -236,6 +236,52 @@ def before_run
     hop_wait
   end
 
+  label def configure_metrics
+    metrics_config = postgres_server.metrics_config
+    metrics_dir = metrics_config[:metrics_dir]
+    vm.sshable.cmd("mkdir -p #{metrics_dir}")
+    vm.sshable.cmd("tee #{metrics_dir}/config.json > /dev/null", stdin: metrics_config.to_json)
+
+    metrics_service = <<SERVICE
+[Unit]
+Description=PostgreSQL Metrics Collection
+After=postgresql.service
+
+[Service]
+Type=oneshot
+User=ubi
+ExecStart=/home/ubi/postgres/bin/metrics-collector #{metrics_dir} #{metrics_dir}/config.json
+StandardOutput=journal
+StandardError=journal
+SERVICE
+    vm.sshable.cmd("sudo tee /etc/systemd/system/postgres-metrics.service > /dev/null", stdin: metrics_service)
+
+    metrics_interval = metrics_config[:interval] || "15s"
+
+    metrics_timer = <<TIMER
+[Unit]
+Description=Run PostgreSQL Metrics Collection Periodically
+
+[Timer]
+OnBootSec=30s
+OnUnitActiveSec=#{metrics_interval}
+AccuracySec=1s
+
+[Install]
+WantedBy=timers.target
+TIMER
+    vm.sshable.cmd("sudo tee /etc/systemd/system/postgres-metrics.timer > /dev/null", stdin: metrics_timer)
+
+    vm.sshable.cmd("sudo systemctl daemon-reload")
+
+    when_initial_provisioning_set? do
+      vm.sshable.cmd("sudo systemctl enable --now postgres-metrics.timer")
```

Comment:
The logic (and the area of work) is in close proximity of `configure_prometheus`. How about (after moving the above to image generation) combining both of them under this label `configure_metrics`?

Reply:
Combined both into a single label